### PR TITLE
feat(components): [scrollbar] add tabindex to scrollbar wrap content

### PR DESCRIPTION
**Pull request by Karolis-Stoncius**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### <samp>🤖 Generated by Copilot at f28a054</samp>

This pull request adds a new `tabindex` prop to the `scrollbar` component and a new `body-tabindex` prop to the `table` component. These props allow users to control the keyboard focus and navigation of the scrollbars in the components. The props are also documented in the corresponding markdown files.

## Related Issue

Fixes #\_\_\_.

### <samp>🤖 Generated by Copilot at f28a054</samp>

*  Add `tabindex` prop to scrollbar component to allow customizing the tabindex of the wrap container ([link](https://github.com/element-plus/element-plus/pull/14320/files?diff=unified&w=0#diff-537bfee7cc12f968cea66b544cebe79f402c021c434fb945be5500047eb69355R53), [link](https://github.com/element-plus/element-plus/pull/14320/files?diff=unified&w=0#diff-14b88eafa1474daa78e77e625fe6e1dfacbbf4f759428f78a41c6a652d1bda31R77-R83), [link](https://github.com/element-plus/element-plus/pull/14320/files?diff=unified&w=0#diff-e27619e12f65c1d00d6cb6b43bc52333fb26fd103f60caa963485c7844cf8496L3-R9))
*  Add `bodyTabindex` prop to table component to allow customizing the tabindex of the body scrollbar's wrap container ([link](https://github.com/element-plus/element-plus/pull/14320/files?diff=unified&w=0#diff-48aed1abfbd975e78df7b5352f8c1321bb18c360349e47559acde0c4346b7123R274), [link](https://github.com/element-plus/element-plus/pull/14320/files?diff=unified&w=0#diff-616bb754e750605914ca6b8e064991f1b1c91c581fd22ce883f77ee438f53c03R66), [link](https://github.com/element-plus/element-plus/pull/14320/files?diff=unified&w=0#diff-616bb754e750605914ca6b8e064991f1b1c91c581fd22ce883f77ee438f53c03R321), [link](https://github.com/element-plus/element-plus/pull/14320/files?diff=unified&w=0#diff-8b6f285bf97996e3543e5cbccdb8a325bc298fdf8c790f650a31f89e72b2ddc3R149), [link](https://github.com/element-plus/element-plus/pull/14320/files?diff=unified&w=0#diff-8b6f285bf97996e3543e5cbccdb8a325bc298fdf8c790f650a31f89e72b2ddc3R277-R280))
*  Bind `tabindex` prop to wrap container element in `scrollbar.vue` template ([link](https://github.com/element-plus/element-plus/pull/14320/files?diff=unified&w=0#diff-e27619e12f65c1d00d6cb6b43bc52333fb26fd103f60caa963485c7844cf8496L3-R9))
*  Bind `bodyTabindex` prop to body scrollbar component in `table.vue` template ([link](https://github.com/element-plus/element-plus/pull/14320/files?diff=unified&w=0#diff-616bb754e750605914ca6b8e064991f1b1c91c581fd22ce883f77ee438f53c03R66))
*  Pass `bodyTabindex` prop to table body component in `table.vue` script ([link](https://github.com/element-plus/element-plus/pull/14320/files?diff=unified&w=0#diff-616bb754e750605914ca6b8e064991f1b1c91c581fd22ce883f77ee438f53c03R321))

### CODE REVIEW

**Comment by Karolis-Stoncius** [comment]
This is the same PR as [this](https://github.com/element-plus/element-plus/pull/13863), I had to delete my fork, so I created a new PR.

**Comment by Karolis-Stoncius** [comment]
@tolking I did the changes you asked [here](https://github.com/element-plus/element-plus/pull/13863), is it okay?

**Comment by tolking** [comment]
Due to many internal components using this ElScrollbar component, it is necessary to take time to verify whether this feature will affect other components.

**Comment by Karolis-Stoncius** [comment]
> Due to many internal components using this ElScrollbar component, it is necessary to take time to verify whether this feature will affect other components.

I can change default `tabindex` value to `undefined` so that it wouldn't affect anything by default

**Comment by Karolis-Stoncius** [comment]
@tolking any updates on this?

File: docs/en-US/component/table.md [hunk 1]
```diff
@@ -271,7 +271,7 @@ table/table-layout
...
+| body-tabindex             | body scrollbar's wrap container tabindex                                                                                                                                                                                                                                    | string / number                                                                                                                                                      | —                                                   | —                                                                                                                       |
```
[the remark this answers is not in the document]
Comment (Karolis-Stoncius) [reply docs/en-US/component/table.md]: Changed it

File: packages/components/table/src/table.vue [hunk 2]
```diff
@@ -318,7 +318,7 @@ export default defineComponent({
       scrollTo,
       setScrollLeft,
       setScrollTop,
-      bodyTabindex: props.bodyTabindex,
+      scrollbarTabindex: props.scrollbarTabindex,
```
Comment (tolking) [inline packages/components/table/src/table.vue:321]:
```suggestion
```

Writing like this looks redundant

File: packages/components/scrollbar/src/scrollbar.vue [hunk 3]
```diff
@@ -1,6 +1,12 @@
 <template>
   <div ref="scrollbarRef" :class="ns.b()">
-    <div ref="wrapRef" :class="wrapKls" :style="style" @scroll="handleScroll">
+    <div
+      ref="wrapRef"
+      :class="wrapKls"
+      :style="style"
+      :tabindex="props.tabindex"
```
Comment (tolking) [inline packages/components/scrollbar/src/scrollbar.vue:7]:
```suggestion
      :tabindex="tabindex"
```

Writing like this looks redundant

File: packages/components/table/src/table.vue (hunk 2 above)
Comment (Karolis-Stoncius) [reply packages/components/table/src/table.vue]: Fixed

File: packages/components/scrollbar/src/scrollbar.vue (hunk 3 above)
Comment (Karolis-Stoncius) [reply packages/components/scrollbar/src/scrollbar.vue]: Fixed

**Review by tolking — APPROVED** [review]

**Comment by Karolis-Stoncius** [comment]
@tolking Hi, when will this be released?

**Comment by Karolis-Stoncius** [comment]
@ryuhangyeong @btea could you review this as well?

**Comment by btea** [comment]
@Karolis-Stoncius Sorry, I think I missed this message before. Now the code seems to have conflicts, do you have time to fix it?

**Comment by Karolis-Stoncius** [comment]
@btea done

File: docs/en-US/component/scrollbar.md [hunk 4]
```diff
@@ -60,6 +60,7 @@ scrollbar/manual-scroll
 | role ^(2.4.0) ^(a11y)             | role of view                                                                                                                    | ^[string]                                                           | —       |
 | aria-label ^(2.4.0) ^(a11y)       | aria-label of view                                                                                                              | ^[string]                                                           | —       |
 | aria-orientation ^(2.4.0) ^(a11y) | aria-orientation of view                                                                                                        | ^[enum]`'horizontal' \| 'vertical'`                                 | —       |
+| tabindex                          | tabindex of wrap container                                                                                                      | ^[number] / ^[string]                                               | —       |
```
Comment (btea) [inline docs/en-US/component/scrollbar.md:63]: Perhaps we could add a version tag to indicate which versions support this configuration.
```suggestion
| tabindex ^(2.8.3)                 | tabindex of wrap container                                                                                                      | ^[number] / ^[string]                                               | —       |
```

File: docs/en-US/component/table.md [hunk 5]
```diff
@@ -285,6 +285,7 @@ table/table-layout
...
 | flexible ^(2.2.1)         | ensure main axis minimum-size doesn't follow the content                                                                                                                                                                                                                   | ^[boolean]                                                                                                                                                           | false                                                                                                                   |
+| scrollbar-tabindex        | body scrollbar's wrap container tabindex                                                                                                                                                                                                                                   | ^[string] / ^[number]                                                                                                                                                | —                                                                                                                       |
```
Comment (Karolis-Stoncius) [inline docs/en-US/component/table.md:288]: @btea should I add it here as well?

Comment (btea) [reply docs/en-US/component/table.md:288]: Yes.

File: docs/en-US/component/scrollbar.md (hunk 4 above)
Comment (Karolis-Stoncius) [reply docs/en-US/component/scrollbar.md]: done

File: docs/en-US/component/table.md (hunk 5 above)
Comment (Karolis-Stoncius) [reply docs/en-US/component/table.md]: done

**Review by btea — APPROVED** [review]
Thanks!

**Pull request merged**